sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -154,6 +230,18 @@ class SQLContext(@transient val sparkContext: SparkContext)
   @transient
   protected[sql] val defaultSession = createSession()
 
+  // Sub class may need to overwrite the following 2 functions custom sql dialect parser support
+
+  protected[sql] def dialectClassName = if (conf.dialect == "sql") {
+    classOf[DefaultSQLDialect].getCanonicalName
+  } else {
+    conf.dialect
+  }
+
+  protected[sql] def resetSQLDialect() {
+    setConf(SQLConf.DIALECT, "sql")
+  }
```

Comment:
Do we really need a special reset function?  There is already logic to clear configuration when testing.